cranelift/codegen/src/isa/x64/mod.rs

```diff
@@ -84,6 +85,18 @@ impl MachBackend for X64Backend {
     fn reg_universe(&self) -> &RealRegUniverse {
         &self.reg_universe
     }
+
+    fn unsigned_add_overflow_condition(&self) -> IntCC {
+        // Unsigned `>=`; this corresponds to the carry flag set on x86, which happens on
+        // overflow of an add.
+        IntCC::UnsignedGreaterThanOrEqual
```

Comment:
It was UnsignedLessThan in the old backend's code; was this a preexisting bug, or is this a bad copy-pasto? I'm not sure it's quite easy to test right now, because there's no support for x86-64 in Spidermonkey, and I'm convinced it's testable with Cranelift's old x86 backend in Spidermonkey.

Reply:
It's unclear to me why it was the way it was; but with the way that things are lowered now, it should be `>=`, as far as I can tell. The documentation for the underlying trait method this fulfills (on `TargetIsa`) says that it is the condition that should be triggered when an add overflows; and that's definitely carry-flag-set on x86 (and aarch64). In any case we will want to carefully check this when we build out the new x86-64 backend.